core/controllers/story_editor.py

```diff
@@ -41,6 +42,10 @@ def get(self, topic_id, story_id):
         if topic is None or story_id not in topic.canonical_story_ids:
             raise self.PageNotFoundException
 
+        self.values.update({
+            'INTERACTION_SPECS': interaction_registry.Registry.get_all_specs()
```

Comment:
This was needed because one of the files needed for image upload in the frontend (ImagePreloaderService) imports ComputeGraphService which needs this constant.

Reply:
I think that merits a comment -- maybe add one?

Also should we file an issue to separate things so that ImagePreloaderService does not require ComputeGraphService? It seems like that's not correct, semantically...

Reply:
The preloader service has just one instance of this where it uses this to compute the BFS path of states and preload images in all states, I think.
So, it is required here as part of the service's functionality.

Reply:
Done, added comment.